### DISCUSSION TITLE
How do I add StackSteps to Application Stage (trying to manually approve change set)

### DISCUSSION THREAD

**Discussion by aws-cdk-noob** [aws/aws-cdk#16982] [Q&A]
Hi All,

I am trying to add Validations to my CDK App so that I can review changes to the application before they are deployed. When looking at the [CDK Docs](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html#validation) I found that adding StackSteps and specifying a ManualReview for the ChangeSet would be ideal. However I cannot find any further examples _anywhere_ of StackSteps being implemented from AWS or in people's individual github repos.

My project is based off of this [AWS Developer Blog post](https://aws.amazon.com/blogs/developer/cdk-pipelines-continuous-delivery-for-aws-cdk-applications/) which uses the new CodePipeline Construct for deploying applications.

In my project structure I have two Typescript files:

- cdkpipelines-demo-pipelines-stack            <-- Contains CdkpipelinesDemoStage (stage contains the demo-stack) & CdkpipelinesDemoPipelineStack (stack contains my CodePipeline)
- cdkpiplines-demo-stack                                   <-- Contains CdkpipelinesDemoStack (stack has my application)

When using AWS' supplied example they give no reference for how to reference the stack of your application. Any ideas for how I'm supposed to supply the stack for stackSteps?

`    pipeline.addStage(appStage, {
      stackSteps: [{
        stack: Stack.of(appStage.service2),  
        pre: [new ManualApprovalStep('Pre-Stack Check')], // Executed before stack is prepared
        changeSet: [new ManualApprovalStep('ChangeSet Approval')], // Executed after stack is prepared but before the stack is deployed
        post: [new ManualApprovalStep('Post-Deploy Check')], // Executed after staack is deployed
      }],
    });`

**Comment by stocks29** [comment]
@aws-cdk-noob , you can add your stack as a property of your stage. For example:

```ts
export class MyStage extends Stage {
  readonly appStack: MyStack;

  constructor(scope: Construct, id: string, stageOpts?: ....) {
    super(scope, id, props);
    const stack = new MyStack(this, 'MyStack', { ... });

    this.appStack = stack;
  }
}
```

Then in your pipeline code you can access it:

```ts
const myStage = new MyStage(....);
myStage.appStack;
```

I hope this helps
